test/test_sparse_csr.py

```diff
@@ -1576,7 +1576,7 @@ def to_sp_block_compressed(t):
                 return res
 
         def ref_half_bfloat16(c, a, b, alpha=None, beta=None, out=None):
-            res = alpha * (a.to_dense() @ b.to_dense()) + beta * c
+            res = alpha * (a.to_dense().to(torch.float32) @ b.to_dense().to(torch.float32)).to(a.dtype) + beta * c
```

Comment:
Let's do it like this: `res = alpha * (a.to_dense().to(torch.float32) @ b) + beta * c`. And in the returning branch instead of `return res` let's do `return res.to(a.dtype)` with a comment.

Reply:
Not critical, actually, so it is up to you :)

Reply:
In C++ both tensors are upcasted to fp32 https://github.com/pytorch/pytorch/blob/3a2cfbb813e19c1648b23079f704829f9997425d/aten/src/ATen/native/sparse/SparseBlasImpl.cpp#L140-L141
Why is your suggestion better?
I'm away from the computer right now. If it's okay with you I'll merge this current version.

Reply:
Nothing really major. Just wanted to avoid casting in `out.copy_(res)`, but it is a super minor thingy anyway...

Reply:
Nothing really major, just wanted to point out that `res....to(a.dtype)` could be avoid if the path `out.copy_(res)` is taken. But it is a really minor thingy, so I am fine with your changes.